### DISCUSSION TITLE
Packaging for AWS

### DISCUSSION THREAD

**Discussion by nskmda** [pypa/setuptools#4084] [Q&A] [answered]
Team,

Is there any way to include dependencies themselves in the source distribution?
I mean, `install` dependencies directly in the source distribution folder to package for AWS deployment?

**Comment by abravalheri** [comment, marked as answer]
I know some tools that allow you to do that like https://github.com/linkedin/shiv or https://github.com/pantsbuild/pex, but that is not something supported in `setuptools` directly.

**Reply by webknjaz** [reply]
Technically even `pip install -t` would do, right?

**Reply by abravalheri** [reply]
Yes, that is true. `pip install -t` can create a folder that, when added to `sys.path`, will have all the dependencies of the package.

**Reply by nskmda** [reply]
I know `pip` supports that.
But I'm trying 2 create some repeatable build process (currently w/PyBuilder). PyBuilder uses `setuptools` under the hood (generates a `setup.py`).
I **GUESS** i might still have 2 do it manually (like creating a custom PyBuilder task which runs `pip`) since `setuptools` can't do that 4 me.

**Reply by webknjaz** [reply]
setuptools is a PEP 517 back-end, not front-end. It's a front-ends responsibility to install things.

**Comment by nskmda** [comment]
> setuptools is a PEP 517 back-end, not front-end. It's a front-ends responsibility to install things.

I didn't ?? that fact @ all.
My ? was if setuptools could support installation to a random target folder, not 2 the Python modules folder. If there was a configuration to specify that target folder by the front-end so that back-end does what's needed.

**Reply by webknjaz** [reply]
Backend does not perform installs but builds dists.

**Reply by nskmda** [reply]
Oh... If that's the meaning of the backend...

Still. I believe I need to build a distro. Just want the dependencies to be included in it.
If `setuptools` can't do that, okay... sigh, tho...

**Reply by abravalheri** [reply]
Setuptools cannot do that.

Probably using additional tools like the ones mentioned in https://github.com/pypa/setuptools/discussions/4084#discussioncomment-7304843 you can achieve that on top of the distribution generated with the help of setuptools.

**Reply by nskmda** [reply]
@abravalheri no, those mentioned won't work for me. 😞

**Reply by abravalheri** [reply]
I hope you find something that works for you. Meanwhile, I will mark this discussion as "solved" (since the answer is "no, setuptools cannot do that").

**Reply by nskmda** [reply]
Yes, sure.
